gradle-buildpack/bin/build

```diff
@@ -0,0 +1,104 @@
+
+# Load user-provided build-time environment variables
+if compgen -G "$env_dir/*" > /dev/null; then
+  for var in "$env_dir"/*; do
+    declare "$(basename "$var")=$(<"$var")"
+  done
+fi
+
+echo "---> Installing JDK"
+
+# If it doesn't exist locally, create a JDK cache layer
+# This makes JDK available to subsequent buildpacks as well.
+if [[ -f $layers_dir/jdk.toml ]]; then
+  cached_jdk_url=$(cat "$layers_dir/jdk.toml" | yj -t | jq -r .metadata.url 2>/dev/null || echo 'JDK TOML parsing failed')
+fi
+if [[ $jdk_url != $cached_jdk_url ]] ; then
+  rm -rf "$layers_dir"/jdk
+  mkdir -p "$layers_dir"/jdk/env
+  wget -q -O - "$jdk_url" | tar pxz -C "$layers_dir"/jdk --strip-components=1
+  echo "launch = true" > "$layers_dir"/jdk.toml
+  echo "build = true" >> "$layers_dir"/jdk.toml
+  echo "cache = true" >> "$layers_dir"/jdk.toml
+  echo -e "[metadata]\n  version = \"$jdk_version\"\n  url = \"$jdk_url\"" >> "$layers_dir"/jdk.toml
+
+  echo "$layers_dir"/jdk > "$layers_dir"/jdk/env/JAVA_HOME
+  if [[ -z $LD_LIBRARY_PATH ]]; then
+    echo "$JAVA_HOME/jre/lib/amd64/server" > $layers_dir/jdk/env/LD_LIBRARY_PATH
+  else
+    echo "$JAVA_HOME/jre/lib/amd64/server:$LD_LIBRARY_PATH" > $layers_dir/jdk/env/LD_LIBRARY_PATH
+  fi
+
+  mkdir -p $layers_dir/jdk/profile.d
+  cat << EOF > $layers_dir/jdk/profile.d/jdk.sh
+export JAVA_HOME=$layers_dir/jdk
+if [[ -z \$LD_LIBRARY_PATH ]]; then
+  export LD_LIBRARY_PATH="\$JAVA_HOME/jre/lib/amd64/server"
+else
+  export LD_LIBRARY_PATH="\$JAVA_HOME/jre/lib/amd64/server:$LD_LIBRARY_PATH"
+fi
+EOF
+fi
+# Set env variables to make jdk accessible
+for var in "$layers_dir"/jdk/env/*; do
+  declare "$(basename "$var")=$(<"$var")"
+done
+export PATH=$layers_dir/jdk/bin:$PATH
+
+target_dir="build/libs"
+
+if [[ ! -d $layers_dir/gradle ]]; then
+  mkdir -p $layers_dir/gradle
+  echo "cache = true" > $layers_dir/gradle.toml
+fi
+ln -s $layers_dir/gradle $HOME/.gradle
+
+if [[ -x gradlew ]]; then
+  echo "---> Running Gradle Wrapper"
+  ./gradlew clean build -x test
+else
+  if [[ -f $layers_dir/maven.toml ]]; then
+    cached_gradle_url=$(cat "$layers_dir/gradle.toml" | yj -t | jq -r .metadata.url 2>/dev/null || echo 'Gradle TOML parsing failed')
+  fi
+  if [[ $gradle_url != $cached_gradle_url ]] ; then
+    echo "---> Installing Gradle"
+    gradle_dir="$layers_dir/gradle"
+    rm -rf "$gradle_dir"
+    mkdir -p "$gradle_dir"
+    wget -q -O gradle.zip "$gradle_url"
+    jar xvf gradle.zip
+    mv "gradle-$gradle_version" "$gradle_dir"
+    rm gradle.zip
+    chmod a+x "$gradle_dir/gradle-$gradle_version/bin/gradle"
+    echo "launch = false" > "$layers_dir"/gradle.toml
+    echo "build = true" >> "$layers_dir"/gradle.toml
+    echo "cache = true" >> "$layers_dir"/gradle.toml
+    echo -e "[metadata]\n  version = \"$gradle_version\"\n  url = \"$gradle_url\"" >> "$layers_dir"/gradle.toml
+  fi
+  export PATH="$PATH:$layers_dir/gradle/gradle-$gradle_version/bin"
+
+  echo "---> Running Gradle"
+  gradle clean build -x test
```

Comment:
I wonder if we should include this in the sample or not. It's pretty rare these days to have a Gradle app w/o a wrapper (unlink Maven where it's still pretty common not to use the wrapper).